pennylane_lightning/core/src/simulators/lightning_qubit/gates/OpToMemberFuncPtr.hpp

```diff
@@ -1089,4 +1168,56 @@ callControlledMatrixOp(ControlledMatrixFuncPtrT<PrecisionT> func,
                 wires, adj);
 }
 
+/**
+ * @brief Call a sparse matrix operation.
+ * @tparam PrecisionT Floating point type for the state-vector.
+ * @tparam IndexT Index type for the sparse matrix.
+ * @param func Function pointer for the gate operation.
+ * @param data Data pointer the gate is applied to
+ * @param num_qubits The number of qubits of the state-vector.
+ * @param row_map_ptr Pointer to the row map of the sparse matrix.
+ * @param col_idx_ptr Pointer to the column index of the sparse matrix.
+ * @param values_ptr Pointer to the values of the sparse matrix.
+ * @param wires Wires the gate applies to.
+ * @param adj If true, we apply the adjoint of the gate.
+ */
+template <class PrecisionT, class IndexT = std::size_t>
+inline void
+callSparseMatrixOp(SparseMatrixFuncPtrT<PrecisionT, IndexT> func,
+                   std::complex<PrecisionT> *data, std::size_t num_qubits,
+                   const IndexT *row_map_ptr, const IndexT *col_idx_ptr,
+                   const std::complex<PrecisionT> *values_ptr,
+                   const std::vector<std::size_t> &wires, bool adj = false) {
+    return func(data, num_qubits, row_map_ptr, col_idx_ptr, values_ptr, wires,
+                adj);
+}
+
+/**
+ * @brief Call a controlled sparse matrix operation.
+ * @tparam PrecisionT Floating point type for the state-vector.
+ * @tparam IndexT Index type for the sparse matrix.
+ * @param func Function pointer for the gate operation.
+ * @param data Data pointer the gate is applied to
+ * @param num_qubits The number of qubits of the state-vector.
+ * @param row_map_ptr Pointer to the row map of the sparse matrix.
+ * @param col_idx_ptr Pointer to the column index of the sparse matrix.
+ * @param values_ptr Pointer to the values of the sparse matrix.
+ * @param controlled_wires Wires controlling the gate.
+ * @param controlled_values Values of the control wires.
+ * @param wires Wires the gate applies to.
+ * @param adj If true, we apply the adjoint of the gate.
+ */
+template <class PrecisionT, class IndexT = std::size_t>
+inline void callControlledSparseMatrixOp(
```

Comment:
We may be able to merge these methods with the existing ops for dense Matrix operations if we abstract away the sparse matrix arguments in a data-class in a follow-up PR

Reply:
I agree.